src/shared/pager.c

```diff
@@ -161,8 +161,8 @@ void pager_close(void) {
                 return;
 
         /* Inform pager that we are done */
-        stdout = safe_fclose(stdout);
-        stderr = safe_fclose(stderr);
+        safe_fclose(stdout);
+        safe_fclose(stderr);
```

Comment:
I really don't like this patch I must say... stdout/stderr now point to an invalidated object. I'd much rather have this point to NULL than to invalid memory...

> Trying to write to stdout/stderr after they are freed would produce a segfault in both cases (whether they are NULL or not). Perhaps doing freopen("/dev/null","w",stdout) might be a better idea.

No. This is not correct. Dereffing a NULL ptr will result in a segfault, always. But accessing free memory != NULL might result in a segfault, but might as well not, there's no rule that the pages are actually made unavailable.

hence, there's quite a difference between between setting stdout/stderr to NULL and just fclose()ing it and leaving it set.